### PR TITLE
feat: Add PolarsAllocator

### PR DESCRIPTION
**Pull request by ruihe774**
This PR adds PolarsAllocator, which relays allocations to the allocator capsule introduced in https://github.com/pola-rs/polars/pull/17817.

### CODE REVIEW

**Comment by ritchie46** [comment]
@ruihe774 CI is failing with the fallback allocator, I believe. Can you take a look?

**Comment by ruihe774** [comment]
> @ruihe774 CI is failing with the fallback allocator, I believe. Can you take a look?

Fixed in d17b225.

File: pyo3-polars/src/alloc.rs [hunk 1]
```diff
@@ -67,14 +67,16 @@ impl PolarsAllocator {
         // Do not allocate in this function,
         // otherwise it will cause infinite recursion.
         self.0.get_or_init(|| {
-            Python::with_gil(|_| {
-                unsafe {
-                    (PyCapsule_Import(ALLOCATOR_CAPSULE_NAME.as_ptr() as *const c_char, 0)
-                        as *const AllocatorCapsule)
-                        .as_ref()
-                }
+            (unsafe { Py_IsInitialized() } != 0)
```
Comment (ritchie46) [inline pyo3-polars/src/alloc.rs]: Should we wait here until python is initialized? In which cases is this `false/true`?

Comment (ruihe774) [reply pyo3-polars/src/alloc.rs]: It is always `true` if we are building a python extension module.

`cargo test` does not build a python extension module but a standalone executable, so it needs special handling.

Comment (ritchie46) [reply pyo3-polars/src/alloc.rs]: Thanks. That context is what I needed. :)

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**